src/vm_stack.zig

```diff
@@ -5303,24 +5336,34 @@ pub const StackVM = struct {
         }
     }
 
-    pub fn resumeInvoke(vm: *VM, module: *ModuleInstance, returns: []Val) anyerror!void {
+    pub fn resumeInvoke(vm: *VM, module: *ModuleInstance, returns: []Val, opts: ResumeInvokeOpts) anyerror!void {
         var self: *StackVM = fromVM(vm);
 
-        std.debug.assert(self.debug_state != null);
-        std.debug.assert(self.debug_state.?.is_invoking);
-
-        const debug_state = &self.debug_state.?;
-        const opcode: Opcode = blk: {
+        var pc: u32 = 0;
+        const opcode: Opcode = if (self.debug_state) |debug_state| blk: {
+            std.debug.assert(debug_state.is_invoking);
+            pc = debug_state.pc;
             for (debug_state.trapped_opcodes.items) |op| {
                 if (op.address == debug_state.pc) {
                     break :blk op.opcode;
                 }
             }
             unreachable; // Should never get into a state where a trapped opcode doesn't have an associated record
+
+        } else blk: {
+            if (metering.enabled) {
+                std.debug.assert(self.meter_state.enabled);
+                pc = self.meter_state.pc;
+                if (opts.meter != metering.initial_meter) {
+                    self.meter_state.meter = opts.meter;
+                }
+                break :blk self.meter_state.opcode;
+            }
+            unreachable;
```

Comment:
Let's add a minor comment as to why this unreachable is here - e.g. "should never resumeInvoke" when there's no function to resume"